'Programming error' when clicking button 2 repeatedly to cycle color history

Not sure what else I can say about this one -- I have button2 configured to cycle color history, and it does this.. but occasionally it also generates this 'programming error' report, with the following backtrace:

```python3
Traceback (most recent call last):
  File "/usr/share/mypaint/gui/document.py", line 553, button_press_cb(self=<gui.document.Document object>, tdw=<TiledDrawWidget object at 0x7fa30966e820 (TiledDrawWidget at 0x2bb6640)>, event=<void>)
return self._dispatch_named_action(None, tdw, event,
   action_name)
# User-configurable forbidding of particular devices
  variables: {'action_name': ('local', u'ColorHistoryPopup')}
  File "/usr/share/mypaint/gui/document.py", line 695, _dispatch_named_action(self=<gui.document.Document object>, win=None, tdw=<TiledDrawWidget object at 0x7fa30966e820 (TiledDrawWidget at 0x2bb6640)>, event=<void>, action_name=u'ColorHistoryPopup')
# Pointer: popup states handle these themselves sanely.
handler.activate(event)
return True
  variables: {'handler.activate': ('local', <bound method State.activate of <gui.stategroup.State object at 0x7fa306c0f950>>), 'event': ('local', <void at 0x3c1e2f0>)}
  File "/usr/share/mypaint/gui/stategroup.py", line 144, activate(self=<gui.stategroup.State object>, action_or_event=<void>)
# currently, we only support mouse buttons being pressed here
assert e.type == gdk.BUTTON_PRESS
# let's just note down what mous button that was
  variables: {'gdk.BUTTON_PRESS': ('global', 4), 'e.type': ('local', <enum GDK_2BUTTON_PRESS of type GdkEventType>)}
AssertionError: 
```

MyPaint version: git e629c7e "Fix exception when releasing modified space bar"
GTK+ version: 3.13.7.9.g0ef0b98 (git) .. also tested with 3.11.6
Distro: Arch Linux, 64bit
Tablet: Monoprice 12x9

Since the latest commit seems relevant, I tried to reproduce the problem using spacebar, but it never occurs, it only occurs when rapidly clicking the actual Button2.

--- a/gui/stategroup.py
+++ b/gui/stategroup.py
@@ -146,9 +146,15 @@ class State (object):
         if action_or_event:
             if not isinstance(action_or_event, gtk.Action):
                 e = action_or_event
-                # currently, we only support mouse buttons being pressed here
-                assert e.type == gdk.BUTTON_PRESS
-                # let's just note down what mous button that was
+                # eat any multiple clicks. TODO should possibly try to e.g.
+                # split a triple click into three clicks in the future.
+                if (e.type == gdk.EventType.DOUBLE_BUTTON_PRESS or
+                    e.type == gdk.EventType.TRIPLE_BUTTON_PRESS):
+                    e.type = gdk.EventType.BUTTON_PRESS
+
+                # currently we only support mouse buttons being single-pressed.
+                assert e.type == gdk.EventType.BUTTON_PRESS
+                # let's just note down what mouse button that was
                 assert e.button
                 if e.button in self.allowed_buttons_masks:
                     self.mouse_button = e.button

